src/pal/src/map/map.cpp

```diff
@@ -2241,7 +2241,9 @@ MAPmmapAndRecord(
     PAL_ERROR palError = NO_ERROR;
     LPVOID pvBaseAddress = NULL;
 
-    pvBaseAddress = mmap(addr, len, prot, flags, fd, offset);
+    off_t adjust = offset & (VIRTUAL_PAGE_MASK);
+
+    pvBaseAddress = mmap(static_cast<char *>(addr) - adjust, len + adjust, prot, flags, fd, offset - adjust);
```

Comment:
@janvorli That sounds reasonable.  Do you have a particular guard name to recommend?  We could just use the larger PAL_MAX_PAGE_SIZE for TARGET_ARM64....  I was using the generic PAL because of something @jkotas had said.  I also liked the idea of this being in amd64 CI testing, since arm64 testing is so light.